homeassistant/components/hatchrest/switch.py

```diff
@@ -0,0 +1,51 @@
+"""Switch platform for Hatch Rest."""
+from __future__ import annotations
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HatchRestEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Hatch Rest control switch."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities([HatchRestSwitch(coordinator)], True)
+
+
+class HatchRestSwitch(HatchRestEntity, SwitchEntity):
+    """Master control switch for Hatch Rest device."""
+
+    @property
+    def name(self) -> str:
+        """Hatch Rest switch name."""
+        return f"{super().device_name} power"
```

Comment:
The naming has changed. Please adjust the code to the new [naming requirements](https://developers.home-assistant.io/docs/core/entity#entity-naming)